src/tools/queryDatasource/datasourceCredentials.test.ts

```diff
@@ -5,33 +5,35 @@ import {
 
 const { resetDatasourceCredentials } = datasourceCredentialsExportedForTesting;
 
-describe('getDatasourceCredentials', () => {
-  const originalEnv = process.env;
+const mocks = vi.hoisted(() => ({
+  mockGetConfig: vi.fn(),
+}));
+
+vi.mock('../../config.js', () => ({
+  getConfig: mocks.mockGetConfig,
+}));
 
+describe('getDatasourceCredentials', () => {
   beforeEach(() => {
     resetDatasourceCredentials();
-    process.env = {
-      ...originalEnv,
-      DATASOURCE_CREDENTIALS: undefined,
-    };
-  });
```

Comment:
I refactored these tests to mock `getConfig` instead of relying on setting and resetting environment variables.